packages/timeline/src/MiniGantt.ts

```diff
@@ -225,34 +216,38 @@ export class MiniGantt extends SVGWidget {
         } : (l, r) => {
             return this.brAxis.scalePos(r[1]) - this.brAxis.scalePos(l[1]);
         });
-        const events = data.filter(d => !d[2]);
-        const ranges = data.filter(d => !!d[2]);
+        const events = data.filter(d => !d[this._endDate_idx]);
+        const ranges = data.filter(d => !!d[this._endDate_idx]);
 
         this.brAxis
             .render()
             ;
         const brAxisBBox = this.brAxis.getBBox();
 
-        this.updateEntityPins(events);
-        const upperBBox = this.gUpper.node().getBBox();
-        const upperHeight = Math.max(upperBBox.height, 100);
-        const lowerHeight = height - upperHeight;
-        const gUpperTransY = upperHeight - 5;
-        this.gUpper
-            .attr("class", "gUpper")
-            .attr("transform", `translate(0,${gUpperTransY})`)
-            ;
-        this.gLower
-            .attr("class", "gLower")
-            .attr("transform", `translate(0,${upperHeight})`)
-            ;
+        const upperContentHeight = this.updateEntityPins(events);
+        const upperAxisHeight = 28; // this.gUpperAxis.node().getBBox().height;
```

Comment:
The upperAxis might have labels which are staggered and its height may well be larger than 28?

Reply:
I'm thinking that dynamically determining the height of the upper axis at that point would require the Axis synchronously calculating its height. Any suggestions on how to accomplish that?

Reply:
You can ask it for its height once its been rendered